subsys/bluetooth/host/gatt.c

```diff
@@ -5780,66 +5778,66 @@ static uint8_t ccc_load(const struct bt_gatt_attr *attr, uint16_t handle,
 static int ccc_set(const char *name, size_t len_rd, settings_read_cb read_cb,
 		   void *cb_arg)
 {
-	if (IS_ENABLED(CONFIG_BT_SETTINGS)) {
-		struct ccc_store ccc_store[CCC_STORE_MAX];
-		struct ccc_load load;
-		bt_addr_le_t addr;
-		ssize_t len;
-		int err;
-		const char *next;
-
-		settings_name_next(name, &next);
-
-		if (!name) {
-			LOG_ERR("Insufficient number of arguments");
-			return -EINVAL;
-		} else if (!next) {
-			load.addr_with_id.id = BT_ID_DEFAULT;
-		} else {
-			unsigned long next_id = strtoul(next, NULL, 10);
+#if defined(CONFIG_BT_SETTINGS)
```

Comment:
Developer note: I had to use `#if defined` because CONFIG_BT_SETTINGS_CCC_STORE_MAX only exists if CONFIG_BT_SETTINGS is defined. So IS_ENABLED cannot be used here.

Reply:
I would prefer setting `CCC_STORE_MAX` to `0`[^1] or `CONFIG_BT_SETTINGS_CCC_STORE_MAX` if enabled. It's better for code analysis to reduce the number of ifdefs.

[^1]: I'm pretty sure our tool chain allows zero-length arrays. If not, then setting it to `1` is fine.

Reply:
Sounds good. So you mean:

```c
#if defined(CONFIG_BT_SETTINGS_CCC_STORE_MAX)
#define CCC_STORE_MAX CONFIG_BT_SETTINGS_CCC_STORE_MAX
#else /* defined(CONFIG_BT_SETTINGS_CCC_STORE_MAX) */
#define CCC_STORE_MAX 0
#endif /* defined(CONFIG_BT_SETTINGS_CCC_STORE_MAX) */
```

Reply:
Done.